tests/e2e/e2e_suite_test.go

```diff
@@ -57,240 +53,61 @@ type testContextType struct {
 	imageTag        string
 	arguments       string
 	namespace       string
+	noCleanup       bool
 }
 
 var _ = Describe("dcgm-exporter-e2e-suite", func() {
-	When("DCGM exporter is deployed on kubernetes", Ordered, func() {
+	Context("DCGM exporter is deployed on kubernetes", Ordered, func() {
 		// Init global suite vars
 		var (
-			kubeClient *framework.KubeClient
-			helmClient *framework.HelmClient
-
-			labels = map[string]string{
-				"e2eRunID": runID.String(),
+			kubeClient    *framework.KubeClient
+			helmClient    *framework.HelmClient
+			testRunLabels = map[string]string{
+				e2eRunIDLabel: runID.String(),
 			}
-
-			helmReleaseName string
-			dcgmExpPod      *corev1.Pod
-			workloadPod     *corev1.Pod
 		)
 
-		BeforeAll(func(ctx context.Context) {
-			if testContext.kubeconfig == "" {
-				_, _ = fmt.Fprintln(GinkgoWriter, "kubeconfig parameter is empty. Defaulting to ~/.kube/config")
-			}
-
-			if len(testContext.chart) == 0 {
-				Fail("chart parameter is empty")
-			}
-
-			shouldResolvePath()
+		if testContext.kubeconfig == "" {
+			_, _ = fmt.Fprintln(GinkgoWriter, "kubeconfig parameter is empty. Defaulting to ~/.kube/config")
+		}
 
-			kubeConfigShouldExists()
+		if len(testContext.chart) == 0 {
+			Fail("chart parameter is empty")
+		}
 
-			k8sConfig := shouldCreateK8SConfig()
+		shouldResolvePath()
 
-			kubeClient = shouldCreateKubeClient(k8sConfig)
-
-			helmClient = shouldCreateHelmClient(k8sConfig)
-		})
+		kubeConfigShouldExists()
 
-		AfterAll(func(ctx context.Context) {
-			_, _ = fmt.Fprintln(GinkgoWriter, "Clean up: starting")
-
-			shouldUninstallHelmChart(helmClient, helmReleaseName)
-			shouldCleanupHelmClient(helmClient)
+		k8sConfig := shouldCreateK8SConfig()
 
-			shouldDeleteNamespace(ctx, kubeClient)
+		kubeClient = shouldCreateKubeClient(k8sConfig)
 
-			_, _ = fmt.Fprintln(GinkgoWriter, "Clean up: completed")
-		})
+		helmClient = shouldCreateHelmClient(k8sConfig)
 
-		It("should create namespace", func(ctx context.Context) {
-			shouldCreateNamespace(ctx, kubeClient, labels)
+		BeforeAll(func(ctx context.Context) {
+			shouldCreateNamespace(ctx, kubeClient, testRunLabels)
 		})
```

Comment:
Just out of curiosity why are these not part of `BeforeAll` anymore?

Reply:
@rohit-arora-dev  Is it an old comment?  BeforeAll now creates a namespace as part of the BeforeAll. 

Global variables such as kubeClient, helmClient, and testRunLabels initialize once in the Context, which is the container (https://onsi.github.io/ginkgo/#organizing-specs-with-container-nodes. ) for e2e tests. After global variables are initialized, they are available for child specs, like BeforeAll, etc.